src/context.rs

```diff
@@ -105,10 +106,83 @@ fn test_should_add_a_service() {
         }
     }
 
-    let service = ServiceStub;
-    let mut foo = Context::new(false, Some("localhost".to_owned()), None, None);
+    mod add_service {
+        use super::super::*;
 
-    assert_eq!(foo.services.is_empty(), true);
-    foo.add_service(Box::new(service));
-    assert_eq!(foo.services.is_empty(), false);
+        #[test]
+        fn test_should_increase_number_of_services() {
+            let service = super::ServiceStub;
+            let context = Context::shared(false, Some("localhost".to_owned()), None, None);
+            let mut locked_context = context.lock().unwrap();
```

Comment:
I'm not a fan either. The macro approach sounds good. I'll give it a try.

I was trying something like:
```rust 
// in tests module
fn setup() -> (Context, ServiceStub) {
    let service = ServiceStub;
    let context = Context::shared(false, Some("localhost".to_owned()), None, None);
    let mut locked_context = context.lock().unwrap();
    (locked_context, service)
}

// in add_service
[test]
fn a_test() {
    let (context, service) = &mut super::setup();
    // and the test continues
}
```

The error I got was: 
```
src/context.rs:113:10: 113:24 error: mismatched types:
 expected `context::Context`,
    found `std::sync::mutex::MutexGuard<'_, context::Context>`
(expected struct `context::Context`,
    found struct `std::sync::mutex::MutexGuard`) [E0308]
src/context.rs:113         (locked_context, service)
```